dev/AnimatedVisualPlayer/TestUI/AnimatedVisualPlayerPage.xaml

```diff
@@ -15,87 +15,66 @@
-                IsEnabled="False"
-                AutomationProperties.Name="ToZeroKeyframeAnimationProgressTextBox"
-                Text="None"
-                RelativePanel.Below="ToZeroKeyframeAnimationPlayButton"/>
-            <Button x:Name="FromOneKeyframeAnimationPlayButton"
-                AutomationProperties.Name="FromOneKeyframeAnimationPlayButton"
-                Click="FromOneKeyframeAnimationPlayButton_Click"
-                RelativePanel.Below="ToZeroKeyframeAnimationProgressTextBox"/>
-            <TextBox x:Name="FromOneKeyframeAnimationProgressTextBox"
-                IsEnabled="False"
-                AutomationProperties.Name="FromOneKeyframeAnimationProgressTextBox"
-                Text="None"
-                RelativePanel.Below="FromOneKeyframeAnimationPlayButton"/>
-            <Button x:Name="ReverseNegativePlaybackRateAnimationPlayButton"
-                AutomationProperties.Name="ReverseNegativePlaybackRateAnimationPlayButton"
-                Click="ReverseNegativePlaybackRateAnimationPlayButton_Click"
-                RelativePanel.Below="FromOneKeyframeAnimationProgressTextBox"/>
-            <TextBox x:Name="ReverseNegativePlaybackRateAnimationTextBox"
-                IsEnabled="False"
-                AutomationProperties.Name="ReverseNegativePlaybackRateAnimationTextBox"
-                Text="None"
-                RelativePanel.Below="ReverseNegativePlaybackRateAnimationPlayButton"/>
-            <Button x:Name="ReversePositivePlaybackRateAnimationPlayButton"
-                AutomationProperties.Name="ReversePositivePlaybackRateAnimationPlayButton"
-                Click="ReversePositivePlaybackRateAnimationPlayButton_Click"
-                RelativePanel.Below="ReverseNegativePlaybackRateAnimationTextBox"/>
-            <TextBox x:Name="ReversePositivePlaybackRateAnimationTextBox"
-                IsEnabled="False"
-                AutomationProperties.Name="ReversePositivePlaybackRateAnimationTextBox"
-                Text="None"
-                RelativePanel.Below="ReversePositivePlaybackRateAnimationPlayButton"/>
-            <TextBox x:Name="HittestingTextBox"
-                IsEnabled="False"
-                AutomationProperties.Name="HittestingTextBox"
-                Text="None"
-                RelativePanel.Below="ReversePositivePlaybackRateAnimationTextBox"/>
-            <Button x:Name="FallenBackButton"
-                AutomationProperties.Name="FallenBackButton"
-                Click="FallenBackButton_ClickAsync"
-                RelativePanel.Below="HittestingTextBox"/>
-            <TextBox x:Name="FallenBackTextBox"
-                IsEnabled="False"
-                AutomationProperties.Name="FallenBackTextBox"
-                Text="None"
-                RelativePanel.Below="FallenBackButton"/>
-            <controls:AnimatedVisualPlayer
+        <Grid.Resources>
+            <Style TargetType="TextBox">
+                <Setter Property="IsEnabled" Value="False" />
```

Comment:
If you want a read-only TextBox, you can use IsReadOnly. Alternatively, use a TextBlock instead of a TextBox.

Reply:
The TextBoxes are used by the automation tests to indicate state changes. TextBlocks don't fire state changes. By setting IsEnabled=False I get something that looks like a TextBlock (i.e. you can't select it or attempt to edit it) but with the automation behavior that we need. A TextBox that is IsReadOnly looks like a regular text block, and if IsEnabled=False the IsReadOnly is redundant. I'll add a comment.


---
In reply to: [343825146](https://github.com/microsoft/microsoft-ui-xaml/pull/1558#discussion_r343825146) [](ancestors = 343825146)

Reply:
Comments added to the code to explain this choice.

Reply:
Thanks for the explanation. That makes sense. I didn't think of the change notifications.

---
In reply to: [344318126](https://github.com/microsoft/microsoft-ui-xaml/pull/1558#discussion_r344318126) [](ancestors = 344318126,343825146)